gator/fragments.py

```diff
@@ -1,64 +1,59 @@
 """Retrieve and count the contents of a file"""
 
 from pathlib import Path
-import re
+
+import commonmark
 
 from gator import util
 
 FILE_SEPARATOR = "/"
 
-# References:
-# https://stackoverflow.com/questions/18568105/how-match-a-paragraph-using-regex
-# https://stackoverflow.com/questions/13531204/how-to-match-paragraphs-in-text-with-regx
-
-CODE_FENCE_MARKER = "```"
-GATORGRADER_REPLACEMENT = "GATORGRADER_REPLACEMENT"
-PARAGRAH_RE = r"(.+?\n\n|.+?$)"
-SECTION_MARKER = "#"
-
 NEWLINE = "\n"
 NOTHING = ""
 SPACE = " "
 
-DOUBLE_NEWLINE = NEWLINE * 2
-
 
-def is_paragraph(candidate):
-    """Determines if a writing candidate is a paragraph"""
-    # remove whitespace surrounding candidate paragraph
-    candidate = candidate.strip()
-    # if the paragraph is a markdown header, it is not a paragraph
-    if candidate.startswith(SECTION_MARKER):
-        return False
-    # if the paragraph is a fenced code block, it is not a paragraph
-    if candidate.startswith(CODE_FENCE_MARKER):
-        return False
-    # there may be other edge cases that should be added here in the
-    # future -- what other structures look like paragraphs but should
-    # not be?
-    # if nothing has returned by now, the candidate must be a paragraph
-    return True
+def is_single_subnode(subnode):
+    """Determines if a node is single"""
+    return (
+        # Excludes the following subnode types
+        subnode.t == "text"
+        or subnode.t == "thematic_break"
+        or subnode.t == "html_block"
+        or subnode.t == "code_block"
+        or subnode.t == "code"
+        or subnode.t == "softbreak"
+        or subnode.t == "linebreak"
+    )
 
 
-def get_paragraphs(contents, blank_replace=True):
+def get_paragraphs(contents):
     """Retrieves the paragraphs in the writing"""
-    # use a replacement to handle a string with just spaces
-    if blank_replace is True:
-        contents = contents.replace(SPACE, NOTHING)
-    # replace a single newline with a blank space, respecting double newlines
-    contents = contents.replace(DOUBLE_NEWLINE, GATORGRADER_REPLACEMENT)
-    contents = contents.replace(NEWLINE, SPACE)
-    contents = contents.replace(GATORGRADER_REPLACEMENT, DOUBLE_NEWLINE)
-    pattern = re.compile(PARAGRAH_RE)
-    paragraphs = pattern.findall(contents)
-    # disregard all of the section headers in markdown
-    matching_paragraphs = []
-    # iterate through all potential paragraphs and gather
-    # those that match the standard for legitimacy
-    for paragraph in paragraphs:
-        if is_paragraph(paragraph) is True:
-            matching_paragraphs.append(paragraph)
-    return matching_paragraphs
+    # Calls the CommonMark parser
```

Comment:
I'm not sure this comment is necessary. The review team (@finneyj2, @durcij) should have a discussion on whether inline documentation comments should be used or not. If they should, perhaps more should be added. Please work with the implementation team to determine what those comments should be, if you decide they are needed!